### PR TITLE
ADR-38: OCSP Peer Verification

### PR DESCRIPTION
**Pull request by tbeets**
Technical documentation of the OCSP Peer Verification server feature.

### CODE REVIEW

**Comment by tbeets** [comment]
@wallyqs I'd like to get this ADR merged as I have an updated ADR for Cert Store feature right behind it.  Can you review?

File: adr/ADR-38.md [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+
+The `ocsp_peer` configuration option may be specified in short or long forms.
+
+###### Short form
+The short form is a boolean value:
+
+| `ocsp_cache`            | OCSP cache behavior                                                           |
+|-------------------------|-------------------------------------------------------------------------------|
+| `true` (default, unset) | Is enabled; equivalent to long form with `type: local` and otherwise defaults |
+| `false`                 | Is disabled; equivalen to long form with `type: none`                         |
```
Comment (wallyqs) [inline adr/ADR-38.md:160]:
```suggestion
| `false`                 | Is disabled; equivalent to long form with `type: none`                         |
```

File: adr/ADR-38.md [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+In the NATS Server configuration file, the `ocsp_cache` configuration option may be used to explicitly enable a
+server-scoped OCSP response cache. Such cache will be used for all TLS listeners enabled for OCSP Peer Verification (as above).
+
+> Note: If `ocsp_cache` is configured, but no TLS listeners are enabled for OCSP Peer Verification, the NATS Server will
+> not initialize a cache. If `ocsp_cache` is absent, but one or more TLS listeners are enabled for OCSP Peer Verification, the NATS Server
+> _will_ initialize a local cache with default settings. This is equivalent to `ocsp_cache: true`.
+
+#### Defaults, short, and long form
+
+The `ocsp_peer` configuration option may be specified in short or long forms.
```
Comment (wallyqs) [inline adr/ADR-38.md:152]:
```suggestion
The `ocsp_cache` configuration option may be specified in short or long forms.
```

**Review by wallyqs — APPROVED** [review]
LGTM!

**Pull request merged**